tests/StringHelperTest.php

```diff
@@ -120,6 +120,31 @@ public function testTruncateWords(): void
         $this->assertEquals('это строка с          неожиданными…', StringHelper::truncateWords(' это строка с          неожиданными пробелами ', 4));
     }
 
+    #[DataProvider('provideTruncateWordsByLength')]
+    public function testTruncateWordsByLength(string $expected, string $input, int $length, string $trimMarker = '…', string $encoding = 'UTF-8'): void
+    {
+        $this->assertSame($expected, StringHelper::truncateWordsByLength($input, $length, $trimMarker, $encoding));
+    }
+
+    public static function provideTruncateWordsByLength(): array
+    {
+        return [
+            'basic truncation' => ['Do you like drink…', 'Do you like drink coffee at work?', 20],
+            'string shorter than limit should return as-is' => ['Short text', 'Short text', 20],
+            'string exactly at limit should return as-is' => ['Exact length text', 'Exact length text', 17],
+            'custom trim marker' => ['Do you like drink!!!', 'Do you like drink coffee at work?', 23, '!!!'],
+            'multibyte characters' => ['это тестовая…', 'это тестовая multibyte строка', 15],
+            'no spaces (single word) - should break the word' => ['verylongwo…', 'verylongword', 11],
+            'very short limit with marker' => ['A…', 'A long sentence', 2],
+            'limit with the same length as marker' => ['…', 'Some text', 1],
+            'empty string' => ['', '', 10],
+            'spaces that exceed limit should truncate to empty' => ['', '     ', 3],
+            'text with trailing spaces should be trimmed' => ['Hello world', 'Hello world   ', 15],
+            'multiple words that fit exactly' => ['Hel…', 'Hello world', 6],
+            'space at start should add marker' => ['…', ' Hello', 4],
```

Comment:
```suggestion
            'space at start should add marker' => ['Hel…', ' Hello', 4],
```

Maybe trim spaces on both sides? @samdark WDYT?